operator/manager/manager.go

```diff
@@ -438,7 +446,38 @@ func (m *Manager) AfterAll(ctx context.Context, stage *core.Stage) error {
 		Status:    m.Status,
 		Users:     m.Users,
 	}
-	return t.do(ctx, stage)
+
+	// run the teardown, this updates the build/stages
+	if err := t.do(ctx, stage); err != nil {
+		logger = logger.WithError(err)
+		logger.Warnln("manager: cannot teardown build")
+		return err
+	}
+
+	updater := &updater{
+		Builds:  m.Builds,
+		Events:  m.Events,
+		Repos:   m.Repos,
+		Steps:   m.Steps,
+		Stages:  m.Stages,
+		Webhook: m.Webhook,
+	}
+
+	stages, err := m.Stages.ListSteps(noContext, stage.BuildID)
+	if err != nil {
+		logger.WithError(err).Warnln("manager: cannot get stages")
+		return err
+	}
+
+	// send a final update for build success/failure
+	if isBuildComplete(stages) {
```

Comment:
I think all of this code can be moved to the teardown function, and can be run after the github status is updated (and after isBuildComplete is checked)
https://github.com/drone/drone/blob/master/operator/manager/teardown.go#L181:L189

Reply:
I would have done that but the teardown object doesn't know about the webhook provider. Should I just pass the webhook provider to teardown?

Reply:
yes, I would update the teardown object to accept the webhook provider.